File: app/src/main/java/com/example/bykov/roomdbapp/database/IUserDataSource.java
package com.example.bykov.roomdbapp.database;

import com.example.bykov.roomdbapp.model.User;

import java.util.List;

import io.reactivex.Flowable;

/**
 * Created by bykov on 13.06.2018.
 */

public interface IUserDataSource {
    Flowable<User> getUserById(int userId);

    Flowable<List<User>> getAllUsers();

    void insertUser(User... users);

    void updateUser(User... users);

    void deleteUser(User user);

    void deleteAllUsers();
}

File: app/src/main/java/com/example/bykov/roomdbapp/database/IUserDataSource.kt
package com.example.bykov.roomdbapp.database

import com.example.bykov.roomdbapp.model.User
import io.reactivex.Flowable

/**
 * Created by bykov on 13.06.2018.
 */
interface IUserDataSource {
    fun getUserById(userId: Int): Flowable<User>
    fun getAllUsers(): Flowable<List<User>>
    fun insertUser(user: User)
    fun updateUser(user: User)
    fun deleteUser(user: User)
    fun deleteAllUsers()
}